src/game/logic/object/weapon.cpp

```diff
@@ -155,6 +148,172 @@ Weapon *WeaponStore::Allocate_New_Weapon(const WeaponTemplate *tmpl, WeaponSlotT
     return new Weapon(tmpl, wslot);
 }
 
+WeaponStore::~WeaponStore()
+{
+    Delete_All_Delayed_Damage();
+
+    for (unsigned int i = 0; i < m_weaponTemplateVector.size(); i++) {
+        WeaponTemplate *tmplate = m_weaponTemplateVector[i];
+
+        if (tmplate != nullptr) {
+            tmplate->Delete_Instance();
+        }
+    }
+
+    m_weaponTemplateVector.clear();
+}
+
+void WeaponStore::Handle_Projectile_Detonation(const WeaponTemplate *tmplate,
+    const Object *source_obj,
+    const Coord3D *victim_pos,
+    int bonus_condition,
+    bool do_damage) const
+{
+    Weapon *weapon = g_theWeaponStore->Allocate_New_Weapon(tmplate, WEAPONSLOT_PRIMARY);
+    weapon->Load_Ammo_Now(source_obj);
+    weapon->Fire_Projectile_Detonation_Weapon(source_obj, victim_pos, bonus_condition, do_damage);
+    weapon->Delete_Instance();
+}
+
+void WeaponStore::Create_And_Fire_Temp_Weapon(
+    const WeaponTemplate *tmplate, const Object *source_obj, const Coord3D *victim_pos)
+{
+    if (tmplate) {
+        Weapon *weapon = g_theWeaponStore->Allocate_New_Weapon(tmplate, WEAPONSLOT_PRIMARY);
```

Comment:
allocateNewWeapon exists in rts.map as public: class Weapon near * thiscall WeaponStore::allocateNewWeapon(class WeaponTemplate const near *,enum  WeaponSlotType)const which matches the function prototype in ZH WB.
makeWeapon has a different prototype (no WeaponSlotType and it returns Weapon rather than Weapon *) so I think allocateNewWeapon is the correct name.